components/raft_log_engine/src/engine.rs

```diff
@@ -297,8 +298,34 @@ impl RaftEngine for RaftLogEngine {
         Ok(())
     }
 
-    fn gc(&self, raft_group_id: u64, _from: u64, to: u64) -> Result<usize> {
-        Ok(self.0.compact_to(raft_group_id, to) as usize)
+    fn gc(&self, raft_group_id: u64, from: u64, to: u64) -> Result<usize> {
+        self.batch_gc(vec![RaftLogGCTask {
+            raft_group_id,
+            from,
+            to,
+        }])
+    }
+
+    fn batch_gc(&self, tasks: Vec<RaftLogGCTask>) -> Result<usize> {
+        let mut batch = self.log_batch(tasks.len());
+        let mut old_first_index = Vec::with_capacity(tasks.len());
+        for task in &tasks {
+            batch
+                .0
+                .add_command(task.raft_group_id, Command::Compact { index: task.to });
```

Comment:
According to my understanding,`Compact` can only remove the previous range [-∞, to], but not the range selected by gc, am i right?

Reply:
Yes.